sycl/test-e2e/Annotated_usm/annotated_usm.cpp

```diff
@@ -0,0 +1,173 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// E2E tests for annotated USM allocation functions
+
+#include <sycl/sycl.hpp>
+
+#include <complex>
+#include <numeric>
+
+// clang-format on
+
+using namespace sycl::ext::oneapi::experimental;
+using namespace sycl::ext::intel::experimental;
+using alloc = sycl::usm::alloc;
+
+// Single test instance consisting of (i) calling malloc function `f` (ii)
+// checking usm location of returned pointer  (iii) calling free
+#define TEST_USM_KIND(f, kind, args...)                                        \
+  {                                                                            \
+    auto ap = f(args);                                                         \
+    assert(ap.get() != nullptr &&                                              \
+           sycl::get_pointer_type(ap.get(), Ctx) == kind);                     \
+    free(ap, q);                                                               \
+  }
+
+// Single test instance consisting of (i) calling malloc function `f` (ii)
+// checking alignment Note that the annotated_ptr is not freed in alignment
```

Comment:
The USM tests use a different approach to test alignment, see https://github.com/intel/llvm/blob/1bc609bded02295676039def09c23e411f217346/sycl/test-e2e/USM/alloc_functions.cpp#L36

Can you use the same technique, which is a bit more robust?

Reply:
test updated